src/providers/EtherspotContextProvider.tsx

```diff
@@ -364,6 +364,41 @@ const EtherspotContextProvider = ({
     [sdk, accountAddress]
   );
 
+  const getRatesByTokenAddresses = useCallback(
+    async (chainId: number, tokenAddresses: string[]) => {
+      let tokens = tokenAddresses.filter((address) => address !== null);
+      if (!sdk || !tokens.length) return null;
+      try {
+        const rates: RateData = await sdk.fetchExchangeRates({ tokens, chainId });
+        if (!rates.errored && rates.items.length) {
+          let rateByAddress: Record<string, number> = {};
+          rates.items.forEach((rate) => {
+            rateByAddress[rate.address] = rate.usd;
+          });
+          return rateByAddress;
+        }
+      } catch (error) {}
+    },
+    [sdk]
+  );
+
+  const getRatesByNativeChainId = useCallback(
+    async (chainId: number) => {
+      if (!sdk) return null;
+      try {
+        const rates: RateData = await sdk.fetchExchangeRates({
+          tokens: ['0x0000000000000000000000000000000000000000'],
```

Comment:
is this right and works for all native tokens?

Reply:
also 0x can be replaced by zero address constant

Reply:
Yeah so I asked Vignesh for the native token thing he said 0x0000000000000000000000000000000000000000 will do